app/pages/activity/components/activityForm.tsx

```diff
@@ -0,0 +1,150 @@
+import { useState } from "react";
+import { Button } from "@/components/ui/button";
+import { Input } from "@/components/ui/input";
+import { Label } from "@/components/ui/label";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@/components/ui/select";
+import { Plus } from "lucide-react";
+import { ActivityRequestTypeEnum } from "@/api/models/activity-request";
+import { UserActivityResponse } from "@/api/models/user-activity-response";
+import { useAddActivityMutation } from "../hooks/useAddActivityMutation";
+import EmoticonPicker from "./emoticonPicker";
+import { getIconComponent } from "../utils/iconsHelper";
+import React from "react";
+
+interface ActivityFormProps {
+  activities?: UserActivityResponse;
+  userId: number;
+}
+
+const ActivityForm: React.FC<ActivityFormProps> = ({ activities, userId }) => {
+  const [selectedActivity, setSelectedActivity] = useState<string | null>(null);
+  const [customActivityName, setCustomActivityName] = useState("");
+  const [customActivityType, setCustomActivityType] =
+    useState<ActivityRequestTypeEnum>("CLOCK");
+  const [duration, setDuration] = useState<number | null>(null);
+
+  const { mutate: addActivity, error: addError } = useAddActivityMutation();
```

Comment:
addError słaba nazwa zmiennej, sugeruje że to jest fukncja a nie jest . lepiej nazwać to addActivityError